lib/kafka/produce_operation.rb

```diff
@@ -137,6 +137,9 @@ def handle_response(response)
               delay: ack_time - message.create_time,
             })
           end
+        rescue Kafka::MessageSizeTooLarge
+          @logger.error "Message is too large when writing to #{topic}/#{partition}"
+          @buffer.clear_messages(topic: topic, partition: partition)
```

Comment:
The data loss implications of this are terrifying for us. Could the buffer clearing at least be configurable? Even `CorruptMessage` does not cause the buffer to be cleared when it is raised.